snuba/consumers/consumer.py

```diff
@@ -68,7 +68,11 @@ class BytesInsertBatch(NamedTuple):
     rows: Sequence[bytes]
     origin_timestamp: Optional[datetime]
 
-    def __reduce_ex__(
+    # TODO: Argument 1 of "__reduce_ex__" is incompatible with supertype "object"; supertype defines the argument type as "SupportsIndex"
+    # This is coming from the type hints builtins.pyi:
+    # if sys.version_info >= (3, 8):
+    #    def __reduce_ex__(self, __protocol: SupportsIndex) -> str | tuple[Any, ...]: ...
+    def __reduce_ex__(  # type: ignore
         self, protocol: int
     ) -> Tuple[Any, Tuple[Sequence[Any], Optional[datetime]]]:
         if protocol >= 5:
```

Comment:
Still not ideal but how about:

```suggestion
    def __reduce_ex__(
        self, protocol: SupportsIndex
    ) -> Tuple[Any, Tuple[Sequence[Any], Optional[datetime]]]:
        if int(protocol) >= 5:
```

Reply:
Because then I get an error at `if int(protocol) >= 5`

![Screen Shot 2022-07-13 at 3 50 05 PM](https://user-images.githubusercontent.com/2727124/178819648-33e5c29e-0f8f-409b-a5ff-452011987aee.png)

Reply:
I'm not sure I get why you would see this error if the LHS was being cast to an int.

Reply:
Me either, I couldn't figure it out.

Reply:
This actually worked for me when I tried it locally. Mind double checking?